### ISSUE TITLE
Writing event at begining for single ct()

### ISSUE THREAD

**Issue by stuwilkins** [NSLS-II/bluesky#700]
One use currently at 23-ID is to take XPCS data by taking a single data point. We would like to be able to read that data while collecting the data. Now SWMR is in AD and we have support in `filestore` this is possible. However, for a single data point, the entry to "databroker" is written at the end meaning we have no FS entry to get the data until it is finished.

## Expected Behavior

We need a way to have an entry placed in the FS and corresponding machinery so the databroker can retrieve the data during an acquisition. 

## Current Behavior

As above, this does not work.

**Comment by danielballan** [comment]
I have good news. There is a [custom plan already in your IPython profile](https://github.com/NSLS-II-CSX/xf23id1_profiles/blob/master/profile_collection/startup/95-plans.py#L102) for this purpose.

**Comment by stuwilkins** [comment]
Thanks @danielballan, My bad for not double checking, I was aware this fixed the CTRL-C problem but not the issue above, and @ambarb and @cmazzoli weren't aware too. However that is embarrassing for us! ;-) Very sorry! Someday I would like a test setup for all these things so we don't have to rely on the (very few) hot commissioning days. Now I have a lab detector this might now be possible. 

While this is a 23-ID ism what is the plan for this to migrate upwards? It seems like XPCS is not the only use case for this as I am sure other people want similar functionality? I would be curious on your thoughts.......

**Comment by danielballan** [comment]
I agree it's likely to pop up elsewhere. We haven't settled on a good way to parameterize it. Do we add an extra keyword argument to _every_ plan exposing a switch for this choice? Do we create variants of every plan that have this functionality instead of the "normal" one of throwing away interrupted data points? None jumps out as an obviously "best" way, so we haven't moved forward on this. Let's discuss here and try to decide on a way forward.
